src/architecture/network.js

```diff
@@ -395,7 +395,92 @@ Network.prototype = {
     // Remove the node from this.nodes
     this.nodes.splice(index, 1);
   },
-
+  
+  /**
+   * Checks whether a given mutation is possible, returns an array of candidates to use for a mutation when it is.
+   *
+   * @param {mutation} method [Mutation method](mutation)
+   *
+   * @returns {false | object[]} candidates to use for a mutation. Entries may be arrays containing paris when appropriate.
+   *
+   * @example
+   *
+   * const network = new architect.Perceptron(2,3,1)
+   *
+   * network.possible(mutation.SUB_NODE) // returns an array of nodes that can be removed
+   */
+  possible: function mutationIsPossible(method) {
+    const self = this
+    let candidates = []
+    switch (method) {
+      case mutation.SUB_NODE:
+        candidates = _.filter(this.nodes, function(node) { return (node.type !== 'output' && node.type !== 'input') }) // assumes input & output node 'type' has been set
```

Comment:
This is a VERY _risky_ assumption to make right now.

Reply:
This assumption is baked into various parts of codebase which is probably the source of existing bugs like #101 

We need to look into creating better "type" setting for `node`